libs/client-sdk/src/data_sources/data_source_event_handler.cpp

```diff
@@ -145,22 +144,22 @@ DataSourceEventHandler::MessageStatus DataSourceEventHandler::HandleMessage(
             status_manager_.SetError(
                 DataSourceStatus::ErrorInfo::ErrorKind::kInvalidData,
                 kErrorParsingDelete);
-            return DataSourceEventHandler::MessageStatus::kInvalidMessage;
+            return MessageStatus::kInvalidMessage;
         }
-        auto res = boost::json::value_to<
-            tl::expected<DataSourceEventHandler::DeleteData, JsonError>>(
+        auto res = boost::json::value_to<tl::expected<DeleteData, JsonError>>(
             boost::json::parse(data));
         if (res.has_value()) {
-            handler_.Upsert(context_, res.value().key,
-                            ItemDescriptor(res.value().version));
-            return DataSourceEventHandler::MessageStatus::kMessageHandled;
+            handler_.Upsert(
+                context_, res.value().key,
```

Comment:
Explicit construction of the tombstone strong typedef is now needed since the constructor only accepts those instead of raw `uint64_t`.